phoenix-core/src/main/java/org/apache/phoenix/util/ServerUtil.java

```diff
@@ -417,4 +418,27 @@ public static <T> Throwable getExceptionFromFailedFuture(Future<T> f) {
         }
         return t;
     }
+
+    /**
+     * Returns true if HBase namespace exists, else returns false
+     * @param admin HbaseAdmin Object
+     * @param schemaName Phoenix schema name for which we check existence of the HBase namespace
+     * @return true if the HBase namespace exists, else returns false
+     * @throws SQLException If there is an exception checking the HBase namespace
+     */
+    public static boolean isHBaseNamespaceAvailable(Admin admin, String schemaName) throws IOException{
+        boolean namespaceExists = false;
+        try{
+            String[] hbaseNamespaces = admin.listNamespaces();
+            for(String namespace : hbaseNamespaces){
```

Comment:
nit: can we replace this with a utility method?

Reply:
Done